cnf/Makegap.in

```diff
@@ -683,3 +674,15 @@ gapmpi.o: @srcdir@/system.h @srcdir@/gasman.h @srcdir@/objects.h \
           @srcdir@/vars.h @srcdir@/stats.h @srcdir@/gapmpi.h 
 	$(CC) -I. $(MPI_CFLAGS) -DCONFIG_H $(CFLAGS) -o gapmpi.o -c @srcdir@/gapmpi.c
 
```

Comment:
To expand on this: `Makegap.in` is a generated file, just like `configure.out`. We are intentionally violating the rule that generated files should not be in the repository (whether that violation is a good idea or not is another matter, and irrelevant for this PR). Both files are generated by `Makefile`. Part of what this PR does is to turn some of the rather opaque and hard to read / debug / change `echo` commands in `Makefile` into text in `Makegap.bottom` which is simply copied to `Makegap.in`. That makes it a bit easier to understand what's going on, and to tweak those build rules if necessary.